From: sunshineinabox <aqemail@gmail.com>
Date: Wed, 30 Sep 2026 00:10:00 -0700
Subject: [PATCH] libinput: keep the output udev names for a device

Default OutputName to the WL_OUTPUT name from udev instead of clearing
it.
---
--- a/src/backends/libinput/device.cpp
+++ b/src/backends/libinput/device.cpp
@@ -896,6 +896,11 @@
     }
 }
 
+QString Device::defaultOutputName() const
+{
+    return QString::fromLocal8Bit(libinput_device_get_output_name(m_device));
+}
+
 void Device::setOutputName(const QString &name)
 {
 #ifndef KWIN_BUILD_TESTING
--- a/src/backends/libinput/device.h
+++ b/src/backends/libinput/device.h
@@ -683,10 +683,7 @@
      * Only sets the output name, for config loading purposes
      */
     void setConfigOutputName(const QString &name);
-    QString defaultOutputName() const
-    {
-        return {};
-    }
+    QString defaultOutputName() const;
 
     void setOutputUuid(const QString &uuid);
     QString defaultOutputUuid() const
